.github/workflows/build_and_test.yml

```diff
@@ -99,7 +99,7 @@ jobs:
     env:
       RUSTC_WRAPPER: cachepot
     steps:
-    # cache 
+    - uses: actions/checkout@v2
```

Comment:
This would check out before installing the dependencies, wouldn't that mean the dependencies get re-installed evey time?

Reply:
Actually, the thing out of the cache scoop will be re-installed because every runner will run on a clean system. I move `checkout` to the first line in order to support the `hashFiles` operator in the cache key, otherwise `hashFiles` will return none.

Reply:
emmmm I may add some comments next time to clarify the strange change 😂